paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java

```diff
@@ -124,16 +125,28 @@ public List<Long> listAllIds() {
         }
     }
 
+    /** Check TableScheme is be modified. */
+    public void checkTableSchema(TableSchema oldSchema, TableSchema newSchema) {
+        boolean isCommon =
+                oldSchema.version() == newSchema.version()
+                        && oldSchema.id() == newSchema.id()
```

Comment:
I don't think we need to compare schema id and options.

Reply:
Okay, if there's no change in the entire logic, I'll remove the schema id and options